data/test/scenarios/test_assert.cfg

```diff
@@ -27,3 +27,276 @@
         {RETURN ([not][true][/true][/not])}
     [/event]
 )}
+
+# Same as the above three but using the Lua unit_test module
+{GENERIC_UNIT_TEST "test_lua_assert" (
+    [event]
+        name = start
+        [lua]
+            code=<<
+                unit_test.assert(true, 'assert true')
+                unit_test.pass()
+            >>
+        [/lua]
+    [/event]
+)}
+{GENERIC_UNIT_TEST "test_lua_assert_fail" (
+    [event]
+        name = start
+        [lua]
+            code=<<
+                unit_test.assert(false, 'assert false')
+                unit_test.pass()
+            >>
+        [/lua]
+    [/event]
+)}
+{GENERIC_UNIT_TEST "test_lua_assert_fail_two" (
+    [event]
+        name = start
+        [lua]
+            code=<<
+                unit_test.assert(true, 'assert true')
+                unit_test.pass_if(false, 'return false')
+            >>
+        [/lua]
+    [/event]
+)}
+
```

Comment:
I think there should be a few tests for sanity-checking the framework with multiple events. From the current implementation of finish() we can be sure this works, but it seems worth the sanity-check in case someone refactors the framework.

```
# Sanity check that a fail() in a fired event causes the test to fail
{GENERIC_UNIT_TEST "test_lua_fail_in_fired_event" (
    [event]
        id = fail_event
        [lua]
            code=<<
                unit_test.fail()
            >>
        [/lua]
    [/event]
    [event]
        name = start
        [fire_event]
            id = fail_event
        [/fire_event]
        [lua]
            code=<<
                unit_test.pass()
            >>
        [/lua]
    [/event]
)}
```